src/crimson/osd/pg.cc

```diff
@@ -936,8 +950,30 @@ PG::submit_transaction(
     projected_log.add(entry);
   }
 
+  std::vector<PGBackend::pg_shard_should_send> shards;
+  std::vector<pg_shard_t> to_push_clone;
+  const auto &pg_shards = peering_state.get_acting_recovery_backfill();
+  for (auto &pg_shard : pg_shards) {
+    if (pg_shard == pg_whoami) {
+      continue;
+    }
+    auto &hoid = obc->obs.oi.soid;
+    if (should_send_op(pg_shard, hoid)) {
+      shards.emplace_back(pg_shard, true);
+    } else {
+      shards.emplace_back(pg_shard, false);
+      if (_new_clone && is_missing_on_peer(pg_shard, hoid)) {
```

Comment:
Can we just pass _new_clone into PGBackend::submit_transaction and have it be responsible for figuring out whether it needs to be sent?  I'd prefer that PG::submit_transaction not be determining which shards get the transaction.

Reply:
Yeah, will do